### ISSUE TITLE
Destructured Props in Functional Component

### ISSUE THREAD

**Issue by skipjack** [milesj/babel-plugin-typescript-to-proptypes#45]
Thanks for your work on this. I just have a question for which I didn't see an answer in the README or open/closed issues, apologies if I missed it. Would the following inline type definitions in the following example be picked up and converted to prop types?

``` ts
const MyComponent = ({
  title: String,
  count: Number,
  ...
}) => {
  return null
}
```

If not, would you be open to supporting it?

**Comment by milesj** [comment]
@skipjack That syntax seems a bit invalid, do you mean something liek this?

```
const MyComponent = (props: {
  title: string,
  count: number,
  ...
}) => {
  return null
}
```

**Comment by skipjack** [comment]
Good point! 🤦  Forgot that the aliasing operator within destructuring conflicts with type assignment (both `:`). Yeah so either what you had...

``` ts
const MyComponent = (props: {
  title: string,
  count: number
}) => {
  return null
}
```

Where you'd have to reference `props.title` or `props.count`. Or the following which I [verified here][1]...

``` ts
const MyComponent = ({
  title = 'Untitled',
  count = 0
}: {
  title: string,
  count: number
}) => {
  return null
}
```

Not sure if I love either, without aliasing in the way my first example (where I fixed the capitalization and added defaults) would be the most concise and clean imo but it is what it is. Anyway curious whether the other alternatives would be viable with this plugin?

[1]: https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCoASMFmgY7p7ehCTkVOle4jUMdRLYTqCc8LEZzCZmoNJODPHFZZXVtZYYkAAeRJTInDQS8po+rf40gnjbDKv8LqD2jpbYoACqAEoAMsK7sUmxkGSCc+VVQQuaTwVb1UBrDYULY7PagbgUZLJH6QbYmJAECjuMigZEMVDsJzCFLNXxtajBFQITiCGhmACiqwYiBqkAAao4KAxpO5QABeUAAIgAEjBoBhwPAyNFBcVqYJ4O4AuLmAAKBlMlnuDmUbnuACUlLAOAaHjIXkQUucIPYNG+bjMzHYDGYkH4sGg6kS3PgehEXkFjOZrMEgqSOMF4AcqAA5MIAHKQCSCoIqADy3zIu0Ebq0kGOqEcClc0jwoCclp0oyw+hQCEQMOEV3tEil0REBg4OO6LDYXG4QWC8poBEo2wAsvoga7+aBVQBvRQlEqoeS8gUxh7bNeQbgx3hL5eRNECgAMigAvvlQIvl2gd9faWPmAe78fUNfOOxENIfpf9fyAB8N6HiU8qKpAyrwGqAAGAAk86rqg7hXqACHvheMGGhewSjhkqCToI06QAu97IZA14xjg+ixu6HDbNeACsoAXvqQA
